### PR TITLE
pkp/pkp-lib#7014 Bring DOIs into the core application and refactor to support deposit status

### PR DESCRIPTION
**Pull request by ewhanson**

### CODE REVIEW

File: classes/plugins/PubObjectsExportPlugin.inc.php [hunk 1]
```diff
@@ -391,10 +408,11 @@ abstract public function getExportDeploymentClassName();
      * @param string $filter
      * @param Context $context
      * @param bool $noValidation If set to true no XML validation will be done
+     * @param $outputErrors null|mixed
```
Comment (NateWr) [inline classes/plugins/PubObjectsExportPlugin.inc.php:411]: Add a message documenting that errors will be placed into this variable after the operation is complete.

File: cypress/tests/integration/Doi.spec.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		cy.get('#article-doi-management button:contains("Bulk Actions")').click({multiple: true});
+		cy.get('button#openBulkMarkRegistered').click();
+
+		// Confirm assignment
+		cy.get('div[data-modal="bulkActions"] button:contains("Mark DOIs registered")').click();
+		cy.get('.app__notifications').contains('Items successfully marked registered', {timeout:20000});
+
+		cy.get(`#list-item-submission-${submissionId} .pkpBadge`).contains('Registered');
+	});
+})
```
Comment (NateWr) [inline cypress/tests/integration/Doi.spec.js:92]: If it's annoyiing to set up separate DOI tests in pkp-lib, you can add custom cypress commands to move some code into pkp-lib. If you look for `commands.js` you''ll see some existing commands defined. These are then called with `cy.myCustomCommand()`.

With that you should be able to get most of this into pkp-lib and maintain less code across three apps.

File: locale/en_US/api.po [hunk 3]
```diff
@@ -41,6 +41,12 @@ msgstr "You do not have permission to delete this server."
 msgid "api.contexts.404.contextNotFound"
 msgstr "The server you requested was not found."
 
+msgid "api.dois.400.contextsNotMatched"
+msgstr "The DOI you requested is not part of this journal."
```
Comment (NateWr) [inline locale/en_US/api.po:45]: journal to server

File: locale/en_US/locale.po [hunk 4]
```diff
@@ -32,6 +32,12 @@ msgstr "Unassigned"
 msgid "common.queue.long.active"
 msgstr "Active"
 
+msgid "doi.submit.noContext"
```
Comment (NateWr) [inline locale/en_US/locale.po:35]: I think you can use `api.contexts.404.contextNotFound` instead for this use-case.

File: locale/en_US/manager.po [hunk 5]
```diff
@@ -146,6 +149,13 @@ msgstr "Display on Homepage"
 msgid "manager.setup.numAnnouncementsHomepage.description"
 msgstr "How many announcements to display on the homepage. Leave this empty to display none."
 
+msgid "manager.setup.enableDois.description"
+msgstr "Assign Digital Object Identifiers (DOIs) to preprints and galleys."
+
+msgid "doi.manager.settings.doiObjectsRequired"
+msgstr "Select which items will be assigned a DOI. Most journals assign DOIs to articles, but you may wish to assign DOIs to all of the published items."
```
Comment (NateWr) [inline locale/en_US/manager.po:156]: journals to preprint servers

Comment (NateWr) [reply locale/en_US/manager.po:156]: "all of the published items" -> "preprint galleys as well"

Comment (NateWr) [reply locale/en_US/manager.po:156]: articles -> preprints

File: pages/dois/DoisHandler.inc.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        $templateMgr->setState(['components' => $stateComponents]);
+
+        $templateMgr->assign([
+            'pageTitle' => __('plugins.pubIds.doi.manager.displayName'),
+            'displayPreprintsTab' => !empty($enabledDoiTypes),
+        ]);
+
+        $templateMgr->display('management/dois.tpl');
+    }
+}
```
Comment (NateWr) [inline pages/dois/DoisHandler.inc.php:95]: Same comment about moving as much of this as possible to pkp-lib.

File: classes/migration/upgrade/v3_4_0/I7014_DoiMigration.inc.php [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\doi\Doi;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I7014_DoiMigration extends Migration
```
[the remark this answers is not in the document]
Comment (ewhanson) [reply classes/migration/upgrade/v3_4_0/I7014_DoiMigration.inc.php:14]: Moved to pkp/pkp-lib#7687.

File: cypress/tests/integration/Doi.spec.js (hunk 2 above)
Comment (ewhanson) [reply cypress/tests/integration/Doi.spec.js]: Thanks, that sounds like a good idea. I've spun this off into its own issue (pkp/pkp-lib#7693) along with expanding code coverage for the DOI changes.

**Comment by NateWr** [comment]
Merged manually to `main`.

**Pull request closed without merging**